### PR TITLE
Babel 7 preparation: Switch babel config to JS

### PR DESCRIPTION
**Pull request by IvanGoncharov**
It's the first step towards Babel 7: #1266
Based on this article: https://fatfisz.com/blog/using-babelrc-js-today

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,27 +1,3 @@
...
-    "syntax-async-functions",
-    "syntax-async-generators",
-    "transform-class-properties",
-    "transform-flow-strip-types",
-    "transform-object-rest-spread",
-    ["transform-es2015-classes", {"loose": true}],
-    ["transform-es2015-destructuring", {"loose": true}],
-    ["transform-es2015-spread", {"loose": true}]
-  ]
+  "presets": ["./.babelrc.js"]
```
Comment (IvanGoncharov) [inline .babelrc]: We will remove this file after switching to Babel7

**Comment by leebyron** [comment]
want to resolve conflicts here?

**Comment by IvanGoncharov** [comment]
@leebyron Done

**Comment by IvanGoncharov** [comment]
@leebyron Already rebased + it blocks #1350. Can I merge it?

**Comment by IvanGoncharov** [comment]
@mjmahone Thanks for the review 👍 Merged 🎉

**Pull request merged**